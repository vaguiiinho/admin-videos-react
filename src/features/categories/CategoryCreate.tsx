import {
    Box,
    Paper,
    Typography
} from "@mui/material";
import { useState } from "react";
import { useAppDispatch } from "../../app/hooks";
import { Category, createCategory } from "./categorySlice";
import { CategoryForm } from "./components/CategoryForm";

export function CategoryCreate() {
    const [isDisabled, setIsDisabled] = useState(false)
    const [categoryState, setCategoryState] = useState<Category>({
        id: "",
        name: "",
        description: "",
        isActive: false,
        deleted_at: null,
        createdAt: "",
        updatedAt: "",
    })

    const dispatch = useAppDispatch()

    async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
        e.preventDefault()
        dispatch(createCategory(categoryState))
    }

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target
        setCategoryState({ ...categoryState, [name]: value })
    }

    const handleToggle = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, checked } = e.target;
        setCategoryState({ ...categoryState, [name]: checked });
    };

    return (
        <Box>
            <Paper>
                <Box p={2}>
                    <Box mb={2}>
                        <Typography variant="h4">
                            Create Category
                        </Typography>
                    </Box>
                </Box>
                <CategoryForm
                    isLoading={false}
                    isDisabled={isDisabled}
                    category={categoryState}
                    handleSubmit={handleSubmit}
                    handleChange={handleChange}
                    handleToggle={handleToggle}
                />
            </Paper>
        </Box>
    )
}
