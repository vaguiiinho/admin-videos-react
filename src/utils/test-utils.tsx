import { configureStore } from '@reduxjs/toolkit'
import type { RenderOptions } from '@testing-library/react'
import { render } from '@testing-library/react'
import React, { PropsWithChildren } from 'react'
import { Provider } from 'react-redux'

import type { RootState } from '../app/store'
// As a basic setup, import your same slice reducers
import { apiSlice } from '../features/api/apiSlice'
import { SnackbarProvider } from 'notistack'
import { BrowserRouter } from 'react-router-dom'

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: Partial<RootState>
    store?: ReturnType<typeof configureStore>
}

export function renderWithProviders(
    ui: React.ReactElement,
    extendedRenderOptions: ExtendedRenderOptions = {}
) {
    const {

        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                [apiSlice.reducerPath]: apiSlice.reducer,
            },

        }),
        ...renderOptions
    } = extendedRenderOptions

    const Wrapper = ({ children }: PropsWithChildren) => (
        <Provider store={store}>
            <BrowserRouter>
                <SnackbarProvider>
                    {children}
                </SnackbarProvider>
            </BrowserRouter>
        </Provider>
    )

    // Return an object with the store and all of RTL's query functions
    return {
        store,
        ...render(ui, { wrapper: Wrapper, ...renderOptions })
    }
}