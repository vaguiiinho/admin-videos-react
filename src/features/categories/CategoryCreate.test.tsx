
import { renderWithProviders } from "../../utils/test-utils";
import { CategoryCreate } from "./CategoryCreate";


describe("CategoryCreate", () => {
    it("should render correctly", () => {
        const { asFragment } = renderWithProviders(<CategoryCreate />);
        expect(asFragment()).toMatchSnapshot();
    });
});