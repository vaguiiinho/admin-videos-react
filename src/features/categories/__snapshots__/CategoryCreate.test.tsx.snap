// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCreate should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiBox-root css-19midj6"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
          >
            Create Category
          </h4>
        </div>
      </div>
      <div
        class="MuiBox-root css-19midj6"
      >
        <form>
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-3 css-ipzchf-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r0:"
                    id=":r0:-label"
                  >
                    Name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r0:"
                      name="name"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-81qg8w"
                      >
                        <span>
                          Name *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r1:"
                    id=":r1:-label"
                  >
                    Description
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r1:"
                      name="description"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-81qg8w"
                      >
                        <span>
                          Description *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
            >
              <div
                class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
                  >
                    <span
                      class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary css-zp7cnf-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        aria-label="controlled"
                        class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
                        name="is_active"
                        type="checkbox"
                      />
                      <span
                        class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                  >
                    Active
                  </span>
                </label>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
            >
              <div
                class="MuiBox-root css-10egq61"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                  href="/categories"
                  tabindex="0"
                >
                  Back
                </a>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-dk52k6-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Save
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
